support/Pipelines/Bbh/Inspiral.py

```diff
@@ -35,6 +36,89 @@ def inspiral_parameters(
     """
     id_domain_creator = id_input_file["DomainCreator"]["BinaryCompactObject"]
     id_binary = id_input_file["Background"]["Binary"]
+    left_object = id_binary["ObjectLeft"]
+    right_object = id_binary["ObjectRight"]
+
+    assert (
+        len(left_object.keys()) == 1
+    ), "Expected only one option for 'ObjectLeft'"
+    assert (
+        len(right_object.keys()) == 1
+    ), "Expected only one option for 'ObjectRight'"
+
+    left_object_type, left_object_params = list(left_object.items())[0]
+    right_object_type, right_object_params = list(right_object.items())[0]
+
+    available_solutions = [
+        "KerrSchild",
+        "HarmonicSchwarzschild",
+        "Schwarzschild",
+        "SphericalKerrSchild",
+    ]
+    assert left_object_type in available_solutions, (
+        f"Unknown solution for 'ObjectLeft': '{left_object_type}'. Known"
+        f" solutions are: {available_solutions}"
+    )
+    assert right_object_type in available_solutions, (
+        f"Unknown solution for 'ObjectRight': '{right_object_type}'. Known"
+        f" solutions are: {available_solutions}"
+    )
+
+    mass_left = left_object_params["Mass"]
+    mass_right = right_object_params["Mass"]
+    spin_magnitude_left = np.linalg.norm(
+        left_object_params.get("Spin", [0.0, 0.0, 0.0])
+    )
+    spin_magnitude_right = np.linalg.norm(
+        right_object_params.get("Spin", [0.0, 0.0, 0.0])
+    )
+    total_mass = mass_left + mass_right
+    mass_ratio = max(mass_left, mass_right) / min(mass_left, mass_right)
+    center_left, center_right = id_binary["XCoords"]
+    initial_separation = np.linalg.norm(
+        np.array(center_right) - np.array(center_left)
+    )
+
+    ###########################################################################
+    # Control system parameters
```

Comment:
How about this for structuring the code better than SpEC: You put each of these sections into their own function that returns a dict. Then you call them like this:

```py
params = {
  # ...
}
params.update(
  control_system_parameters(total_mass=total_mass, ...)
)
return params
```